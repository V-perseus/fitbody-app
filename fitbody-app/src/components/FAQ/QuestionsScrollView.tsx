import React, { useState } from 'react'
import { LayoutAnimation, ScrollView } from 'react-native'

//Component
import QuestionAccordion from './QuestionAccordion'
import Questions from './Questions'
import SegueModal from '../SegueModal/SegueModal'

// Types
import { ActiveTab } from '../../screens/tabs/account/FAQ/FAQ'

const ANIMATION_DURATION = 300

interface ICommunityProps {
  section: ActiveTab
}
const Community = ({ section }: ICommunityProps) => {
  const [isShowing, setIsShowing] = useState(false)
  const [linkURL, setLinkURL] = useState('')

  function showModal(url: string) {
    setIsShowing(true)
    setLinkURL(url)
  }

  function closeModal() {
    setIsShowing(false)
    setLinkURL('')
  }

  const slideLayoutAnimation = async () => {
    // just call this function before a state change you want animated
    // requires that list items have key that does not change between rerenders
    const CustomLayoutLinear = {
      duration: ANIMATION_DURATION,
      create: {
        type: LayoutAnimation.Types.easeIn,
        property: LayoutAnimation.Properties.scaleY,
      },
      update: {
        type: LayoutAnimation.Types.easeOut,
      },
    }
    LayoutAnimation.configureNext(CustomLayoutLinear)
  }

  return (
    <>
      <SegueModal
        modalText={'Open link in browser?'}
        noButtonText={'CANCEL'}
        yesButtonText={'VISIT'}
        url={linkURL}
        showModal={isShowing}
        setShowModal={closeModal}
      />
      <ScrollView>
        {Questions[section].map((item, index) => (
          <QuestionAccordion key={`qa${index}`} title={item.title} slideLayoutAnimation={slideLayoutAnimation}>
            {item.description instanceof Function ? item.description({ showModal: showModal }) : item.description}
          </QuestionAccordion>
        ))}
      </ScrollView>
    </>
  )
}

export default Community
