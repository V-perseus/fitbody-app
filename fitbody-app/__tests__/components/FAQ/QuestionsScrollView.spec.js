import React from 'react'
import { cleanup, render, fireEvent } from '@testing-library/react-native'
import renderer from 'react-test-renderer'

import QuestionsScrollView from '../../../src/components/FAQ/QuestionsScrollView'

describe('FAQ QuestionsScrollView', () => {
  afterEach(cleanup)

  it('renders snapshot correctly', () => {
    const tree = renderer.create(<QuestionsScrollView section={'support'} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should show modal when link is pressed', async () => {
    const { queryByText, queryByTestId } = render(<QuestionsScrollView section={'support'} />)
    const question = queryByText(/What are membership plans\? How much do they cost\?/gi)
    expect(question).not.toBeNull()
    await fireEvent.press(question)
    const linkText = queryByTestId('link_btn')
    expect(linkText).toBeDefined()
    await fireEvent.press(linkText)
    expect(queryByText(/Open link in browser\?/i)).toBeDefined()
  })
})
