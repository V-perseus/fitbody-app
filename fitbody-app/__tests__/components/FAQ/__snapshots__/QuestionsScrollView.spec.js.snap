// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQ QuestionsScrollView renders snapshot correctly 1`] = `
[
  <Modal
    animationType="slide"
    backdropPressToClose={true}
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  />,
  <RCTScrollView>
    <View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "borderBottomColor": "#dedede",
            "borderBottomWidth": 1,
            "opacity": 1,
          }
        }
        testID="faq_btn"
      >
        <View>
          <View
            style={
              {
                "borderBottomColor": "#dedede",
                "borderBottomWidth": 1,
              }
            }
          >
            <BVLinearGradient
              colors={
                [
                  0,
                  0,
                ]
              }
              endPoint={
                {
                  "x": 0.5,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                {
                  "x": 0.5,
                  "y": 0,
                }
              }
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 70,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={2}
                style={
                  {
                    "color": "#333333",
                    "flexWrap": "wrap",
                    "fontFamily": "OpenSans",
                    "fontSize": 14,
                    "lineHeight": 20,
                    "paddingBottom": 10,
                    "paddingLeft": 20,
                    "paddingRight": 8,
                    "paddingTop": 10,
                    "width": 675,
                  }
                }
              >
                Is the Fit Body app available worldwide?
              </Text>
              <View
                style={
                  {
                    "justifyContent": "center",
                    "width": 75,
                  }
                }
              >
                <down.svg
                  color="#bbbbbb"
                />
              </View>
            </BVLinearGradient>
          </View>
          <View
            collapsable={false}
            style={{}}
          >
            <View />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "borderBottomColor": "#dedede",
            "borderBottomWidth": 1,
            "opacity": 1,
          }
        }
        testID="faq_btn"
      >
        <View>
          <View
            style={
              {
                "borderBottomColor": "#dedede",
                "borderBottomWidth": 1,
              }
            }
          >
            <BVLinearGradient
              colors={
                [
                  0,
                  0,
                ]
              }
              endPoint={
                {
                  "x": 0.5,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                {
                  "x": 0.5,
                  "y": 0,
                }
              }
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 70,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={2}
                style={
                  {
                    "color": "#333333",
                    "flexWrap": "wrap",
                    "fontFamily": "OpenSans",
                    "fontSize": 14,
                    "lineHeight": 20,
                    "paddingBottom": 10,
                    "paddingLeft": 20,
                    "paddingRight": 8,
                    "paddingTop": 10,
                    "width": 675,
                  }
                }
              >
                Which platforms offer the Fit Body app?
              </Text>
              <View
                style={
                  {
                    "justifyContent": "center",
                    "width": 75,
                  }
                }
              >
                <down.svg
                  color="#bbbbbb"
                />
              </View>
            </BVLinearGradient>
          </View>
          <View
            collapsable={false}
            style={{}}
          >
            <View />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "borderBottomColor": "#dedede",
            "borderBottomWidth": 1,
            "opacity": 1,
          }
        }
        testID="faq_btn"
      >
        <View>
          <View
            style={
              {
                "borderBottomColor": "#dedede",
                "borderBottomWidth": 1,
              }
            }
          >
            <BVLinearGradient
              colors={
                [
                  0,
                  0,
                ]
              }
              endPoint={
                {
                  "x": 0.5,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                {
                  "x": 0.5,
                  "y": 0,
                }
              }
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 70,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={2}
                style={
                  {
                    "color": "#333333",
                    "flexWrap": "wrap",
                    "fontFamily": "OpenSans",
                    "fontSize": 14,
                    "lineHeight": 20,
                    "paddingBottom": 10,
                    "paddingLeft": 20,
                    "paddingRight": 8,
                    "paddingTop": 10,
                    "width": 675,
                  }
                }
              >
                What are membership plans? How much do they cost?
              </Text>
              <View
                style={
                  {
                    "justifyContent": "center",
                    "width": 75,
                  }
                }
              >
                <down.svg
                  color="#bbbbbb"
                />
              </View>
            </BVLinearGradient>
          </View>
          <View
            collapsable={false}
            style={{}}
          >
            <View />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "borderBottomColor": "#dedede",
            "borderBottomWidth": 1,
            "opacity": 1,
          }
        }
        testID="faq_btn"
      >
        <View>
          <View
            style={
              {
                "borderBottomColor": "#dedede",
                "borderBottomWidth": 1,
              }
            }
          >
            <BVLinearGradient
              colors={
                [
                  0,
                  0,
                ]
              }
              endPoint={
                {
                  "x": 0.5,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                {
                  "x": 0.5,
                  "y": 0,
                }
              }
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 70,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={2}
                style={
                  {
                    "color": "#333333",
                    "flexWrap": "wrap",
                    "fontFamily": "OpenSans",
                    "fontSize": 14,
                    "lineHeight": 20,
                    "paddingBottom": 10,
                    "paddingLeft": 20,
                    "paddingRight": 8,
                    "paddingTop": 10,
                    "width": 675,
                  }
                }
              >
                Do you offer a military/student/healthcare discount?
              </Text>
              <View
                style={
                  {
                    "justifyContent": "center",
                    "width": 75,
                  }
                }
              >
                <down.svg
                  color="#bbbbbb"
                />
              </View>
            </BVLinearGradient>
          </View>
          <View
            collapsable={false}
            style={{}}
          >
            <View />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "borderBottomColor": "#dedede",
            "borderBottomWidth": 1,
            "opacity": 1,
          }
        }
        testID="faq_btn"
      >
        <View>
          <View
            style={
              {
                "borderBottomColor": "#dedede",
                "borderBottomWidth": 1,
              }
            }
          >
            <BVLinearGradient
              colors={
                [
                  0,
                  0,
                ]
              }
              endPoint={
                {
                  "x": 0.5,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                {
                  "x": 0.5,
                  "y": 0,
                }
              }
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 70,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={2}
                style={
                  {
                    "color": "#333333",
                    "flexWrap": "wrap",
                    "fontFamily": "OpenSans",
                    "fontSize": 14,
                    "lineHeight": 20,
                    "paddingBottom": 10,
                    "paddingLeft": 20,
                    "paddingRight": 8,
                    "paddingTop": 10,
                    "width": 675,
                  }
                }
              >
                How do I change my membership plan?
              </Text>
              <View
                style={
                  {
                    "justifyContent": "center",
                    "width": 75,
                  }
                }
              >
                <down.svg
                  color="#bbbbbb"
                />
              </View>
            </BVLinearGradient>
          </View>
          <View
            collapsable={false}
            style={{}}
          >
            <View />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "borderBottomColor": "#dedede",
            "borderBottomWidth": 1,
            "opacity": 1,
          }
        }
        testID="faq_btn"
      >
        <View>
          <View
            style={
              {
                "borderBottomColor": "#dedede",
                "borderBottomWidth": 1,
              }
            }
          >
            <BVLinearGradient
              colors={
                [
                  0,
                  0,
                ]
              }
              endPoint={
                {
                  "x": 0.5,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                {
                  "x": 0.5,
                  "y": 0,
                }
              }
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 70,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={2}
                style={
                  {
                    "color": "#333333",
                    "flexWrap": "wrap",
                    "fontFamily": "OpenSans",
                    "fontSize": 14,
                    "lineHeight": 20,
                    "paddingBottom": 10,
                    "paddingLeft": 20,
                    "paddingRight": 8,
                    "paddingTop": 10,
                    "width": 675,
                  }
                }
              >
                How do I cancel my membership?
              </Text>
              <View
                style={
                  {
                    "justifyContent": "center",
                    "width": 75,
                  }
                }
              >
                <down.svg
                  color="#bbbbbb"
                />
              </View>
            </BVLinearGradient>
          </View>
          <View
            collapsable={false}
            style={{}}
          >
            <View />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "borderBottomColor": "#dedede",
            "borderBottomWidth": 1,
            "opacity": 1,
          }
        }
        testID="faq_btn"
      >
        <View>
          <View
            style={
              {
                "borderBottomColor": "#dedede",
                "borderBottomWidth": 1,
              }
            }
          >
            <BVLinearGradient
              colors={
                [
                  0,
                  0,
                ]
              }
              endPoint={
                {
                  "x": 0.5,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                {
                  "x": 0.5,
                  "y": 0,
                }
              }
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 70,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={2}
                style={
                  {
                    "color": "#333333",
                    "flexWrap": "wrap",
                    "fontFamily": "OpenSans",
                    "fontSize": 14,
                    "lineHeight": 20,
                    "paddingBottom": 10,
                    "paddingLeft": 20,
                    "paddingRight": 8,
                    "paddingTop": 10,
                    "width": 675,
                  }
                }
              >
                How do I watch the On Demand classes on my TV?
              </Text>
              <View
                style={
                  {
                    "justifyContent": "center",
                    "width": 75,
                  }
                }
              >
                <down.svg
                  color="#bbbbbb"
                />
              </View>
            </BVLinearGradient>
          </View>
          <View
            collapsable={false}
            style={{}}
          >
            <View />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "borderBottomColor": "#dedede",
            "borderBottomWidth": 1,
            "opacity": 1,
          }
        }
        testID="faq_btn"
      >
        <View>
          <View
            style={
              {
                "borderBottomColor": "#dedede",
                "borderBottomWidth": 1,
              }
            }
          >
            <BVLinearGradient
              colors={
                [
                  0,
                  0,
                ]
              }
              endPoint={
                {
                  "x": 0.5,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                {
                  "x": 0.5,
                  "y": 0,
                }
              }
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 70,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={2}
                style={
                  {
                    "color": "#333333",
                    "flexWrap": "wrap",
                    "fontFamily": "OpenSans",
                    "fontSize": 14,
                    "lineHeight": 20,
                    "paddingBottom": 10,
                    "paddingLeft": 20,
                    "paddingRight": 8,
                    "paddingTop": 10,
                    "width": 675,
                  }
                }
              >
                Still have a question about your membership not answered here?
              </Text>
              <View
                style={
                  {
                    "justifyContent": "center",
                    "width": 75,
                  }
                }
              >
                <down.svg
                  color="#bbbbbb"
                />
              </View>
            </BVLinearGradient>
          </View>
          <View
            collapsable={false}
            style={{}}
          >
            <View />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
]
`;
